x-pack/plugins/apm/public/hooks/use_date_range_redirect.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import qs from 'query-string';
+import { useHistory, useLocation } from 'react-router-dom';
+import { UI_SETTINGS } from '../../../../../src/plugins/data/public';
+import { TimePickerTimeDefaults } from '../components/shared/DatePicker/typings';
+import { useApmPluginContext } from '../context/apm_plugin/use_apm_plugin_context';
+
+export function useDateRangeRedirect() {
```

Comment:
We already have a hook that handles time range `useTimeRange` do you think we could centralize this change in one single hook? does it make sense?

Reply:
They serve different purposes, `useTimeRange` gets the (parsed) range parameters from the url and expects the apm client router to be used, this one checks whether the date range is set in the URL, and can be used by both the APM app and the UX app.

Reply:
Why is it necessary for the date range to be set in the url?

Reply:
discussed with @sqren in AON, we agreed this is ok for now but we will look into whether we can make these query parameters optional and use the kibana date store as the single source of truth.